unified-runtime/source/adapters/hip/usm.cpp

```diff
@@ -37,7 +37,13 @@ urUSMHostAlloc(ur_context_handle_t hContext, const ur_usm_desc_t *pUSMDesc,
     return USMHostAllocImpl(ppMem, hContext, /* flags */ 0, size, alignment);
   }
 
-  return umfPoolMallocHelper(hPool, ppMem, size, alignment);
+  auto UMFPool = hPool->HostMemPool.get();
+  *ppMem = umfPoolAlignedMalloc(UMFPool, size, alignment);
+  if (*ppMem == nullptr) {
+    auto umfErr = umfPoolGetLastAllocationError(UMFPool);
```

Comment:
You should also implement `getProviderNativeError` function (top of this file) to make provider-specific errors be reported properly. I belive the implementation should be the similar to the one in level_zero/usm.cpp

Reply:
I had a quick look, but I think the current HIP provider doesn't report the right name. Although I don't fully understand how all of this works, so I need to look into it further. Maybe that could go in a follow up patch.

Reply:
Oh yeah, I think `get_name()` that's implemented in usm.hpp should return something like "HIP_Provider" so that we can match it properly,